src/components/CheckEditor/CheckEditor.tsx

```diff
@@ -205,24 +189,22 @@ export const CheckEditor = ({ onReturn, checks }: Props) => {
                 Cancel
               </LinkButton>
             </HorizontalGroup>
-            {submissionError && (
-              <div className={styles.submissionError}>
-                <Alert title="Save failed" severity="error">
-                  {`${submissionError.status}: ${
-                    submissionError.data?.msg?.concat(', ', submissionError.data?.err ?? '') ?? 'Something went wrong'
-                  }`}
-                </Alert>
-              </div>
-            )}
           </form>
         </FormProvider>
       </>
+      {error && (
+        <div className={styles.submissionError}>
+          <Alert title="Save failed" severity="error">
+            {error.message ?? 'Something went wrong'}
+          </Alert>
+        </div>
+      )}
       <ConfirmModal
         isOpen={showDeleteModal}
         title="Delete check"
         body="Are you sure you want to delete this check?"
         confirmText="Delete check"
-        onConfirm={onRemoveCheck}
+        onConfirm={onDelete}
```

Comment:
Small thing but I thought I'd start and unify some of the naming so we reference CRUD verbs.